src/Manslice.java

```diff
@@ -0,0 +1,290 @@
+
+    private int BPM;
+
+    private String slice;
+
+    private class Cell {
+
+        @Override
+        public Cell clone() {
+            Cell clone = new Cell();
+            clone.beats = this.beats;
+            clone.openChordRoot = this.openChordRoot;
+            clone.openChordType = this.openChordType;
+            clone.closeChordRoot = this.closeChordRoot;
+            clone.closeChordType = this.closeChordType;
+            clone.openMelody = Arrays.copyOf(this.openMelody,
+                    this.openMelody.length);
+            clone.closeMelody = Arrays.copyOf(this.closeMelody,
+                    this.closeMelody.length);
+            clone.openRhythm = Arrays.copyOf(this.openRhythm,
+                    this.openRhythm.length);
+            clone.closeRhythm = Arrays.copyOf(this.closeRhythm,
+                    this.closeRhythm.length);
+            clone.openBass = Arrays.copyOf(this.openBass, this.openBass.length);
+            clone.closeBass = Arrays.copyOf(this.closeBass,
+                    this.closeBass.length);
+            return clone;
+        }
+
+        private int beats;
+
+        private String openChordRoot, openChordType, closeChordRoot,
+                closeChordType;
+        private String[] openMelody, closeMelody;
+        private String[] openRhythm, closeRhythm;
+        private String[] openBass, closeBass;
+
+    }
+
+    private Queue<Cell> rep;
+
+    private void createNewRep() {
+        this.rep = new Queue2<Cell>();
+    }
+
+    public Manslice() {
+        this.createNewRep();
+        this.rep.enqueue(new Cell());
+    }
+
+    public void setBPM(int bpm) {
+        this.BPM = bpm;
+    }
+
+    public void slice(String slice) {
+        this.slice = slice;
+    }
+
+    public void chordVoice(String voice) {
+        this.chordVoice = voice;
+    }
+
+    public void melodyVoice(String voice) {
+        this.melodyVoice = voice;
+    }
+
+    public void rhythmVoice(String voice) {
+        this.rhythmVoice = voice;
+    }
+
+    public void bassVoice(String voice) {
+        this.bassVoice = voice;
+    }
+
+    public void openCell(int beats) {
+        this.rep.front().beats = beats;
+        this.rep.front().openMelody = new String[beats * 2];
+        this.rep.front().closeMelody = new String[beats * 2];
+        this.rep.front().openRhythm = new String[beats * 2];
+        this.rep.front().closeRhythm = new String[beats * 2];
+        this.rep.front().openBass = new String[beats * 2];
+        this.rep.front().closeBass = new String[beats * 2];
+    }
+
+    public void openChord(String rootNote, String chordType) {
+        this.rep.front().openChordRoot = rootNote;
+        this.rep.front().openChordType = chordType;
+    }
+
+    public void closeChord(String rootNote, String chordType) {
+        this.rep.front().closeChordRoot = rootNote;
+        this.rep.front().closeChordType = chordType;
+    }
+
+    public void openMelody(String[] melody) {
```

Comment:
A recommendation I might have here is that you change the input to varargs:

```suggestion
    public void openMelody(String... melody) {
```

That way, they user doesn't have to construct their own arrays and can list strings directly.